src/cli/init.js

```diff
@@ -0,0 +1,183 @@
+            sourcecred.json files. Otherwise, the command will refuse to
+            overwite pre-existing files and fail.
+
+        --print
+            If provided, sourcecred init will print the project to stdout
+            rather than writing it to sourcecred.json. Mostly used for testing
+            purposes, as a SourceCred instance is only valid if the file is
+            saved as sourcecred.json. If this flag is set, it supercedes
+            --force.
+
+        --help
+            Show this help message and exit, as 'sourcecred help init'.
+
+    Environment variables:
+        SOURCECRED_GITHUB_TOKEN
+            API token for GitHub. This should be a 40-character hex
+            string. Required if you provide GitHub specs.
+
+            To generate a token, create a "Personal access token" at
+            <https://github.com/settings/tokens>. When loading data for
+            public repositories, no special permissions are required.
+            For private repositories, the 'repo' scope is required.
+    `.trimRight()
+  );
+}
+
+function die(std, message) {
+  std.err("fatal: " + message);
+  std.err("fatal: run 'sourcecred help init' for help");
+  return 1;
+}
+
+const initCommand: Command = async (args, std) => {
+  let withForce = false;
+  let printToStdOut = false;
+  let discourseUrl: ?string;
+  let githubSpecs: string[] = [];
+
+  for (let i = 0; i < args.length; i++) {
+    switch (args[i]) {
+      case "--help": {
+        usage(std.out);
+        return 0;
+      }
+      case "--github": {
+        if (++i >= args.length) return die(std, "--github given without value");
+        githubSpecs.push(args[i]);
+        break;
+      }
+      case "--discourse": {
+        if (discourseUrl != undefined)
+          return die(std, "--discourse given multiple times");
+        if (++i >= args.length)
+          return die(std, "--discourse given without value");
+        discourseUrl = args[i];
+        if (!discourseUrl.match(new RegExp("^https?://"))) {
+          return die(
+            std,
+            "invalid discourse url: must start with http:// or https://"
+          );
+        }
+        if (discourseUrl.endsWith("/")) {
+          discourseUrl = discourseUrl.slice(0, discourseUrl.length - 1);
+        }
+        break;
+      }
+      case "--force": {
+        withForce = true;
+        break;
+      }
+      case "--print": {
+        printToStdOut = true;
+        break;
+      }
+      default: {
+        return die(std, `Unexpected argument ${args[i]}`);
+      }
+    }
+  }
+  const dir = process.cwd();
+  const projectFilePath = path.join(dir, "sourcecred.json");
+  const fileAlreadyExists = await fs.exists(projectFilePath);
+  if (fileAlreadyExists && !(withForce || printToStdOut)) {
+    return die(std, `refusing to overwrite sourcecred.json without --force`);
+  }
+
+  const githubToken = Common.githubToken();
+  if (githubToken == null && githubSpecs.length > 0) {
```

Comment:
Nit: I think we can solve this with Flow and a small check.

```js
const githubToken: ?string = Common.githubToken();
if (githubToken) { /* ... */ }
```

As returning an empty string would both be a semantic error (should be `null | undefined` instead).
And `"" == false` anyway so it would retain the current check.

Reply:
When you say "solve this", what problem are you proposing solving? Do you think that comparing the GitHub token to null is a problem? Also, how is flow involved in solving this problem?

While I'm not sure if this is what you're proposing: switching the githubToken check to just evaluate truthiness is a good idea, since we should treat an explicitly empty github token (which you can generate, e.g. via `SOURCECRED_GITHUB_TOKEN="" node bin/sourcecred.js init`) the same as a null github token. So I've made this change.

Reply:
Have no problem with comparing to null, I think the string length check is not a case we should consider here.

My thinking is that:
- `""` is an unexpected, invalid token.
- `null` is an expected case from the `?string` type.
- `undefined` is an expected case from the `?string` type.

Should empty strings cause an issue, but considered equivalent to `null | undefined` `Common.githubToken()` should have returned `null | undefined` instead.

Either way `""` is falsy, so we don't need the `.length > 0` check.